### PR TITLE
PR: Add pyls as a new dependency

### PR DESCRIPTION
**Pull request by ccordoba12**
* Part of issue #7724.
* @CAM-Gerlach, we need an update to our docs for this too.

TODO:

- [x] Update contributing guide

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -203,18 +203,16 @@ a Python version greater than 2.7 or 3.4 (Python <=3.3 is no longer supported).
 * **Python** 2.7 or 3.4+: The core language Spyder is written in and for.
 * **PyQt5** 5.5+: Python bindings for Qt, used for Spyder's GUI.
 * **qtconsole** 4.2.0+: Enhanced Python interpreter.
-* **Rope** 0.9.4+ and **Jedi** 0.11.0+: Editor code completion, calltips
-  and go-to-definition.
-* **Pyflakes**: Real-time code analysis.
+* **Python-language-server**: Editor code completion, calltips
```
Comment (CAM-Gerlach) [inline README.md]: Missing a comma here.

File: README.md [hunk 2]
```diff
@@ -203,18 +203,16 @@ a Python version greater than 2.7 or 3.4 (Python <=3.3 is no longer supported).
...
+  go-to-definition and real-time code analysis
```
Comment (CAM-Gerlach) [inline README.md]: Missing a period.

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Issue created; work forthcoming.

**Comment by bcolsen** [comment]
@ccordoba12 If we don't need Jedi then do we need Numpydocs either? I can't get pyls working on my computer so I can't test it to see.

**Comment by CAM-Gerlach** [comment]
@bcolsen I'd expect we would, unless pyls parses and renders numpydoc docstrings...

**Comment by ccordoba12** [comment]
Jedi is required by pyls, not by us. So it became a third-party dependency and we don't need to have it in our setup.py

We probably need to keep `numpydoc` to make that functionality available through Jedi. When we move our old tests to use pyls I'll find out.

> I can't get pyls working on my computer so I can't test it to see.

Sorry to hear that. So you're not getting code completions in our Editor?

**Comment by bcolsen** [comment]
Thanks @ccordoba12!  I have it working now with pip install.

I only tried with conda install previously. The package installed but it didn't work in spyder.

**Comment by ccordoba12** [comment]
Yeah, the defaults conda channel doesn't have that package yet.

File: README.md [hunk 3]
```diff
@@ -203,18 +203,16 @@ a Python version greater than 2.7 or 3.4 (Python <=3.3 is no longer supported).
...
 * **Sphinx**: Rich text mode for the Help pane.
 * **Pygments** 2.0+: Syntax highlighting for all file types it supports.
 * **Pylint**: Static code analysis.
-* **Pycodestyle**: Real-time code style analysis.
 * **Psutil**: CPU and memory usage on the status bar.
 * **Nbconvert**: Manipulation of notebooks in the Editor.
 * **Qtawesome** 0.4.1+: To have an icon theme based on FontAwesome.
 * **Pickleshare**: Show import completions on the Python consoles.
-* **PyZMQ**: Run introspection services asynchronously.
+* **PyZMQ**: Client for the language server protocol (LSP).
```
Comment (CAM-Gerlach) [inline README.md]: Its rather pointless to define an acronym (LSP) and then never use it...

Comment (ccordoba12) [reply README.md]: That's used in reference in several other IDEs/editors and language implementations. So people who know about the topic will understand what we're talking about.

**Comment by jitseniesen** [comment]
> I have it working now with pip install.

It still does not work for me. I filed issue #7833 to document.

**Pull request merged**